src/NotaryService/NotaryService.cs

```diff
@@ -0,0 +1,409 @@
+            }
+            using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, verifiable, snapshot?.CreateSnapshot(), null, settings, gas))
+            {
+                if (witness.VerificationScript.Length == 0)
+                {
+                    ContractState cs = NativeContract.ContractManagement.GetContract(snapshot, hash);
+                    if (cs is null) return false;
+                    ContractMethodDescriptor md = cs.Manifest.Abi.GetMethod("verify", -1);
+                    if (md?.ReturnType != ContractParameterType.Boolean) return false;
+                    engine.LoadContract(cs, md, CallFlags.ReadOnly);
+                }
+                else
+                {
+                    if (NativeContract.IsNative(hash)) return false;
+                    if (hash != witness.ScriptHash) return false;
+                    Script verificationScript;
+                    try
+                    {
+                        verificationScript = new Script(witness.VerificationScript, true);
+                    }
+                    catch (BadScriptException)
+                    {
+                        return false;
+                    }
+                    engine.LoadScript(verificationScript, initialPosition: 0, configureState: p =>
+                    {
+                        p.CallFlags = CallFlags.ReadOnly;
+                        p.ScriptHash = hash;
+                    });
+                }
+
+                engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
+
+                if (engine.Execute() == VMState.FAULT) return false;
+                if (!engine.ResultStack.Peek().GetBoolean()) return false;
+                fee = engine.GasConsumed;
+            }
+            return true;
+        }
+
+        private void Finalize(Transaction tx)
+        {
+            var prefix = new byte[] { (byte)OpCode.PUSHDATA1, 64 };
+            var notaryWitness = new Witness()
+            {
+                InvocationScript = prefix.Concat(tx.Sign(wallet.GetAccounts().ToArray()[0].GetKey(), neoSystem.Settings.Network)).ToArray(),
+                VerificationScript = new byte[0]
+            };
+            for (int i = 0; i < tx.Signers.Length; i++)
+                if (tx.Signers[i].Account.Equals(NativeContract.Notary.Hash))
+                {
+                    tx.Witnesses[i] = notaryWitness;
+                    break;
+                }
+            neoSystem.Blockchain.Tell(tx);
+        }
+
+        private static void Log(string message, LogLevel level = LogLevel.Info)
+        {
+            Utility.Log(nameof(NotaryService), level, message);
+        }
+
+        public static Props Props(NeoSystem neoSystem, Wallet wallet)
+        {
+            return Akka.Actor.Props.Create(() => new NotaryService(neoSystem, wallet));
+        }
+
+        public class NotaryTask
+        {
+            public bool isSent;
```

Comment:
```suggestion
            public bool IsSent;
```
Same for other fields.